extensions/amp-story/1.0/amp-story.js

```diff
@@ -1285,15 +1302,30 @@ export class AmpStory extends AMP.BaseElement {
 
     if (
       isExperimentOn(this.win, 'amp-story-paywall-exp') &&
-      targetPage.isPaywallProtected()
+      targetPage.isPaywallProtected() &&
+      !this.granted_
     ) {
       if (this.storeService_.get(StateProperty.SUBSCRIPTIONS_DIALOG_STATE)) {
-        // Subscription dialog is already triggered.
+        // Do nothing when the subscription dialog is already triggered.
         return Promise.resolve();
       }
       this.storeService_.dispatch(Action.TOGGLE_SUBSCRIPTIONS_DIALOG, true);
 
-      // TODO(#37285): add SubscriptionService to actually trigger the subscription dialog.
+      this.subscriptionsService_.selectAndActivatePlatform();
```

Comment:
You should simply set `SUBSCRIPTIONS_DIALOG_SET` to `true`, and handle all the work from a listener. Listeners would only be called on change, so if you set it to true multiple times it's a noop.

Reply:
Done.